### PR TITLE
Added the productId GET request

### PR DESCRIPTION
**Pull request by JaimeGSandoval**
view-product-detail

### CODE REVIEW

File: server/index.js [hunk 1]
```diff
@@ -37,6 +38,23 @@ app.get('/api/products', (req, res, next) => {
     .catch(err => next(err));
 });
 
+app.get('/api/products/:productId', (req, res, next) => {
+  const productId = parseInt(req.params.productId, 10);
```
Comment (thebearingedge) [inline server/index.js:42]: You should validate that the `productId` is a positive integer as we've done in prior exercises.

File: server/index.js [hunk 2]
```diff
@@ -37,6 +38,23 @@ app.get('/api/products', (req, res, next) => {
...
+
+  const sql = 'SELECT * FROM "products" WHERE "productId" = $1';
+  const params = [productId];
+
+  db.query(sql, params)
+    .then(result => {
+      const product = result.rows[0];
+      if (!product) {
+        next(new ClientError(`cannot ${req.method} ${req.originalUrl}`, 404));
```
Comment (thebearingedge) [inline server/index.js]: A better message would be something like "cannot find product with id ${productId}"

File: server/index.js [hunk 3]
```diff
@@ -37,6 +38,23 @@ app.get('/api/products', (req, res, next) => {
...
+      }
+      res.json(product);
```
Comment (thebearingedge) [inline server/index.js]: Be sure to include a return statement in your `if` or an `else` to ensure that the server can only respond in one of two ways.

**Review by thebearingedge — CHANGES_REQUESTED** [review]

File: server/index.js (hunk 1 above)
Comment (JaimeGSandoval) [inline server/index.js:42]: Sounds good.

File: server/index.js (hunk 2 above)
Comment (JaimeGSandoval) [reply server/index.js]: Right on, I'll make a clearer message

File: server/index.js (hunk 3 above)
Comment (JaimeGSandoval) [reply server/index.js]: Okay, will do. I should've known that.

**Review by thebearingedge — APPROVED** [review]

**Pull request merged**